### PR TITLE
[rtmidi] conan v2 migration

### PR DESCRIPTION
**Pull request by MartinDelille**
Specify library name and version:  **rtmidi**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — APPROVED** [review]
LGTM

File: recipes/rtmidi/all/conanfile.py [hunk 1]
```diff
@@ -50,34 +58,43 @@ def build_requirements(self):
             self.build_requires("pkgconf/1.7.4")
```
Comment (jwillikers) [inline recipes/rtmidi/all/conanfile.py:58]: Github won't let me suggest changing the previous line here, too, but could you change it to this to only require `pkgconf` when it's not configured directly in Conan?

```suggestion
        if self._with_alsa and not self.conf.get("tools.gnu:pkg_config", default=False, check_type=str):
            self.tool_requires("pkgconf/1.9.3")
```

File: recipes/rtmidi/all/test_package/conanfile.py [hunk 2]
```diff
@@ -1,17 +1,27 @@
-from conans import ConanFile, CMake, tools
+from conan import ConanFile
+from conan.tools.build import can_run
+from conan.tools.cmake import cmake_layout, CMake
 import os
 
 
+# It will become the standard on Conan 2.x
```
Comment (jwillikers) [inline recipes/rtmidi/all/test_package/conanfile.py:7]: Can you remove this comment?

**Review by jwillikers — CHANGES_REQUESTED** [review]
Looks good.

**Review by jwillikers — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**